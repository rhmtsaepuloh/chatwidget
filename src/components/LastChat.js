import React from "react";
import isEmpty from "lodash/isEmpty";

import "../styles/Register.scss";
import Loading from "./Loading";
import LogoPowered from "./LogoPowered";

const LastChat = ({
  onNavigation,
  onCloseChatting,
  loading,
  dataClient,
  lastMessage,
  message,
  uniqueUserId,
}) => {
  const {
    primaryColor,
    secondaryColor,
    buttonColor,
    logo,
    buttonTextColor,
    secondaryTextColor,
  } = dataClient;
  const indexLastMessage = lastMessage.length - 1;
  return (
    <div className='wrapper-register' style={{ background: primaryColor }}>
      <div className='main'>
        <div className='pt-16'>
          <div className='icon-chat h-20 w-20 rounded bg-white mx-auto flex items-center justify-center'>
            <img src={logo} alt='chatajaIcon' className='h-12 w-12 mx-auto' />
          </div>
          <div className='bg-white mx-4 px-4 rounded py-4 mt-8 shadow-md'>
            <p className='text-gray-700 text-sm mb-5 font-bold'>
              Chat Terakhir
            </p>
            <div
              className='grid grid-cols-2 p-3 border border-gray-300 mb-5 rounded hover:bg-gray-400 cursor-pointer'
              onClick={onNavigation}>
              <div className='flex items-center col-span-1'>
                <img
                  src={logo}
                  alt='koperanselicon'
                  className='h-8 w-8 rounded-full mr-3'
                />
                <div className='w-3/4'>
                  <p className='text-base font-bold text-gray-700 m-0'>
                    {isEmpty(lastMessage[indexLastMessage])
                      ? "USER"
                      : lastMessage[indexLastMessage].user.unique_user_id !==
                        uniqueUserId
                      ? "ADMIN"
                      : "USER"}
                  </p>
                  <p className='text-message text-gray-700'>{message()}</p>
                </div>
              </div>
              <div className='col-span-1'>
                <div className='float-right justify-end'>
                  {isEmpty(lastMessage[indexLastMessage]) ? null : (
                    <React.Fragment>
                      <p className='text-xs text-gray-700'>
                        {lastMessage[indexLastMessage].timestamp}
                      </p>
                      <div
                        className='h-4 w-4 rounded-full flex items-center justify-center mx-auto'
                        style={{ background: primaryColor }}>
                        <p className='text-white text-xs'>1</p>
                      </div>
                    </React.Fragment>
                  )}
                </div>
              </div>
            </div>
            <div
              className='text-center px-auto py-3 rounded cursor-pointer hover:bg-red-100'
              style={{ background: buttonColor, color: buttonTextColor }}
              onClick={onCloseChatting}>
              Akhiri Chatting
            </div>
          </div>
          <div
            className='footer-background flex items-center justify-center mt-8 inset-x-0 fixed'
            style={{
              bottom: "0px",
              padding: "10px 0px",
              background: secondaryColor,
            }}>
            <LogoPowered color={secondaryTextColor} />
            <p
              className='text-xs text-center box-border ml-1'
              style={{ color: secondaryTextColor }}>
              Powered by ChatAja
            </p>
          </div>
        </div>
      </div>
      <Loading display={loading} />
    </div>
  );
};

export default LastChat;
