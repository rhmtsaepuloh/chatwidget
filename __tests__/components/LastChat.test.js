/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import LastChat from "../../src/components/LastChat";

const lastMessage = [
  {
    payload: {},
    message: "tes",
    type: "text",
    timestamp: "2020-09-21T04:52:10Z",
    user: {
      fullname: "Client Koperansel",
      avatar_url:
        "https://d1edrlpyc25xu0.cloudfront.net/image/upload/t5XWp2PBRt/1510641299-default_user_avatar.png",
      phone_number: "+6285608849965",
    },
    messageUniqueId: "XXFN1slyO9Km9FXBOuZa1600663929",
    id: 397649556,
  },
];

const emptyLastMessage = [];

const props = {
  onNavigation: jest.fn(),
  onCloseChatting: jest.fn(),
  message: jest.fn(),
  dataClient: {
    hexColor: "#CA1234",
    logo:
      "https://chatwidget-chataja.s3.ap-southeast-1.amazonaws.com/0fcedfb7-1861-42c0-9515-13603ce62901.png",
  },
  uniqueUserId: "user_400f000f319f70f518a53f4d152684b4_bot_36494",
};

describe("render correctly lastchat components", () => {
  test("renders if loading true and last message contain data", () => {
    const wrapper = shallow(
      <LastChat {...props} loading={true} lastMessage={lastMessage} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if loading false and last message contain data", () => {
    const wrapper = shallow(
      <LastChat {...props} loading={false} lastMessage={lastMessage} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if loading true and last message is empty", () => {
    const wrapper = shallow(
      <LastChat {...props} loading={true} lastMessage={emptyLastMessage} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if loading false and last message is empty", () => {
    const wrapper = shallow(
      <LastChat {...props} loading={false} lastMessage={emptyLastMessage} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
