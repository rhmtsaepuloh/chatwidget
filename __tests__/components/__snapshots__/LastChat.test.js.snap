// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly lastchat components renders if loading false and last message contain data 1`] = `ShallowWrapper {}`;

exports[`render correctly lastchat components renders if loading false and last message is empty 1`] = `ShallowWrapper {}`;

exports[`render correctly lastchat components renders if loading true and last message contain data 1`] = `ShallowWrapper {}`;

exports[`render correctly lastchat components renders if loading true and last message is empty 1`] = `ShallowWrapper {}`;
